### PR TITLE
DMEERX-1139 

### PR DESCRIPTION
**Pull request by yuliumitre**
This PR allows the user to select order and/or QuestionnaireResponse, then launch DTR.

To test:
1. For DTR repo, switch to dmeerx-1139 branch as well this repo
2. Select order and/or in-progress QuestionnaireResponse
3. Launch DTR (support for dtr is TBD)

### CODE REVIEW

File: src/components/RequestBox/RequestBox.js [hunk 1]
```diff
@@ -635,12 +662,94 @@ export default class RequestBox extends Component {
...
+    if(order) {
+      appContext += `order=${order}`
+
+      if(coverage) {
+        appContext += `&coverage=${coverage}`
+      }
+    }
+    
+    if(Object.keys(this.state.response).length > 0) {
+      response = `${this.props.ehrUrl}/QuestionnaireResponse/${this.state.response.id}`;
```
Comment (KeeyanGhoreshi) [inline src/components/RequestBox/RequestBox.js]: this can be a local reference

File: src/components/RequestBox/RequestBox.js [hunk 2]
```diff
@@ -635,12 +662,94 @@ export default class RequestBox extends Component {
...
+  }
+
+  buildLaunchLink() {
+    // build appContext and URL encode it
+    let appContext = "";
+    let order = undefined, coverage = undefined, response = undefined;
+
+    if (!this.isOrderNotSelected()) {
+      if (Object.keys(this.state.deviceRequest).length > 0) {
+        order = `${this.props.ehrUrl}/${this.state.deviceRequest.resourceType}/${this.state.deviceRequest.id}`;
```
Comment (KeeyanGhoreshi) [inline src/components/RequestBox/RequestBox.js]: all of these references can be local.  Won't make much of a difference, but this is how it's written in the IG.

Comment (KeeyanGhoreshi) [inline src/components/RequestBox/RequestBox.js:683]: all of these references can be local.  Won't make much of a difference, but this is how it's written in the IG.

File: src/components/RequestBox/RequestBox.js [hunk 3]
```diff
@@ -692,11 +802,15 @@ export default class RequestBox extends Component {
...
               {this.renderPrefetchedResources()}
-            </div>
+            </div> 
+           <div>{this.renderQRInfo()}</div>   
           </div>
         </div>
-        <button className={"submit-btn btn btn-class "} onClick={this.submit}>
-          Submit
+        <button className={"submit-btn btn btn-class "} onClick={this.relaunch} disabled={disableLaunchDTR}>
+          Launch DTR
```
Comment (KeeyanGhoreshi) [inline src/components/RequestBox/RequestBox.js]: This button is a bit ambiguous, perhaps it should say "Relaunch DTR" or have a hover tooltip that says how it is different from clicking on the CRD link.

**Comment by KeeyanGhoreshi** [comment]
Also, minor comment, but the sections seem a bit misaligned.  If it's a bother to fix, don't worry about it.

**Comment by yuliumitre** [comment]
> Also, minor comment, but the sections seem a bit misaligned. If it's a bother to fix, don't worry about it.

@KeeyanGhoreshi can you clarify which sections is misaligned? Screen shots would be helpful. Thanks

**Comment by KeeyanGhoreshi** [comment]
image
The "In progress form" section is indented slightly, so it is misaligned by a bit compared to the other two sections.  Also, it's missing the line underneath the header.

**Review by KeeyanGhoreshi — APPROVED** [review]

**Pull request merged**